### ISSUE TITLE
False positive with `dataclass_transform` + overidden `__init__` + inheritance.

### ISSUE THREAD

**Issue by patrick-kidger** [microsoft/pyright#4738]
```python
from typing_extensions import dataclass_transform

@dataclass_transform()
class Base:
    pass

class A(Base):
    a: int

    def __init__(self, b) -> None:
        self.a = b

class B(A):
    pass

B(b=1)
```
produces
```
error: No parameter named "b" (reportGeneralTypeIssues)
Argument missing for parameter "a" (reportGeneralTypeIssues)
```

Expected behaviour is no error at all. `B.__init__` is inherited from `A` and so it accepts parameter `b`.

Python 3.8.0.
Pyright version 1.1.296 run from the command line.

**Comment by erictraut** [comment]
This is the intended behavior for `dataclass_transform`. An `__init__` method is synthesized if the class doesn't define one. The presence of an `__init__` method in a parent class does not prevent an `__init__` from being synthesized. You would see the same behavior if you marked `A` and `B` as `@dataclass`.
